packages/typescript-react-native-resolver/src/host.ts

```diff
@@ -0,0 +1,265 @@
+}): void {
+  let mode = ResolverLogMode.Never;
+  if (options.traceResolution) {
+    mode = ResolverLogMode.Always;
+  } else if (traceReactNativeModuleResolutionErrors) {
+    mode = ResolverLogMode.OnFailure;
+  }
+  const log = new ResolverLog(mode, traceResolutionLog);
+  host.trace = log.log.bind(log);
+
+  // Ensure that optional methods have an implementation so they can be hooked
+  // for logging.
+  host.directoryExists = host.directoryExists ?? ts.sys.directoryExists;
+  host.realpath = host.realpath ?? ts.sys.realpath;
+  host.getDirectories = host.getDirectories ?? ts.sys.getDirectories;
+
+  changeModuleResolutionHostToLogFileSystemReads(
+    host as ModuleResolutionHostLike
+  );
+
+  const allowedExtensions = [...ExtensionsTypeScript];
+  if (options.checkJs) {
+    allowedExtensions.push(ts.Extension.Js, ts.Extension.Jsx);
+  }
+  if (options.resolveJsonModule) {
+    allowedExtensions.push(ts.Extension.Json);
+  }
+
+  const context: ResolverContext = {
+    host: host as ModuleResolutionHostLike,
+    options,
+    disableReactNativePackageSubstitution,
+    log,
+    platform,
+    platformExtensions: [platform, ...(platformExtensionNames || [])].map(
+      (e) => `.${e}` // prepend a '.' to each name to make it a file extension
+    ),
+    allowedExtensions,
+    replaceReactNativePackageName: createReactNativePackageNameReplacer(
+      platform,
+      disableReactNativePackageSubstitution,
+      log
+    ),
+  };
+
+  host.resolveModuleNames = resolveModuleNames.bind(undefined, context);
+  host.resolveTypeReferenceDirectives = resolveTypeReferenceDirectives.bind(
+    undefined,
+    context
+  );
+}
+
+/**
+ * Resolve a module for a TypeScript program. Prefer type (.d.ts) files and
+ * TypeScript source (.ts[x]) files, as they usually carry more type
+ * information than JavaScript source (.js[x]) files.
+ *
+ * @param context Resolver context
+ * @param moduleName Module name, as listed in the require/import statement
+ * @param containingFile File from which the module was required/imported
+ * @param extensions List of allowed file extensions to use when resolving the module to a file
+ * @returns Resolved module information, or `undefined` if resolution failed.
+ */
+export function resolveModuleName(
+  context: ResolverContext,
+  moduleName: string,
+  containingFile: string,
+  extensions: ts.Extension[]
+): ts.ResolvedModuleFull | undefined {
+  let module: ts.ResolvedModuleFull | undefined = undefined;
+
+  const moduleRef = parseModuleRef(moduleName);
+  const searchDir = path.dirname(containingFile);
+  if (isPackageModuleRef(moduleRef)) {
+    module = resolvePackageModule(context, moduleRef, searchDir, extensions);
+  } else if (isFileModuleRef(moduleRef)) {
+    module = resolveFileModule(context, moduleRef, searchDir, extensions);
+  }
+  if (module) {
+    module.isExternalLibraryImport = !!module.resolvedFileName.match(
+      /[/\\]node_modules[/\\]/
+    );
```

Comment:
[Nit] There's a `test()` function on the RegExp object that we should use instead:
```suggestion
    module.isExternalLibraryImport =
      /[/\\]node_modules[/\\]/.test(module.resolvedFileName);
```